src/omerocrate/uploader.py

```diff
@@ -61,14 +65,18 @@ def check_dependencies(self) -> Self:
             raise ValueError("ROI_Converter_NGFF is required for OmeNgffUploader.")
         return self
 
-    def process_segmentation(self, segmentation_path: Path, image: gateway.ImageWrapper) -> None:
+    def process_segmentation(self, segmentation_path: Path | None, image: gateway.ImageWrapper) -> None:
```

Comment:
Why is this nullable now? I think ideally this would just be not called if there is not segmentation.

Reply:
Thanks, I think it's good to keep the function signatures clean